app/public/static/js/components/keypad.js

```diff
@@ -18,8 +18,10 @@ AFRAME.registerComponent("keypad", {
 
     el.setAttribute(
       "super-keyboard",
-      "imagePath:/static/img/; multipleInputs:true; align: center; maxLength: 4; model:" +
-        data.model
+      "imagePath:/static/img/; multipleInputs:true; align: center; model:" +
+        data.model +
+        "; maxLength: " +
+        password.length
```

Comment:
It is irrelevant to the current task but one optional change you can try is with json like structure just for readability .. so like:
```
el.setAttribute(
      "super-keyboard",
      {
        "imagePath": "/static/img/",
        "multipleInputs":true,
        "align": "center",
        "model": "data.model",
        "maxLength": password.length
      }
    );```